apps/heft/src/operations/runners/LifecycleOperationRunner.ts

```diff
@@ -0,0 +1,131 @@
+    if (watch) {
+      return OperationStatus.Success;
+    }
+
+    const lifecycle: HeftLifecycle = internalHeftSession.lifecycle;
+    const lifecycleLogger: ScopedLogger = internalHeftSession.loggingManager.requestScopedLogger(
+      `lifecycle:${this._options.type}`
+    );
+
+    switch (type) {
+      case 'start': {
+        // We can only apply the plugins once, so only do it during the start operation
+        lifecycleLogger.terminal.writeVerboseLine('Applying lifecycle plugins');
+        await lifecycle.applyPluginsAsync();
+
+        // Run the clean hook
+        if (clean) {
+          const startTime: number = performance.now();
+          const cleanLogger: ScopedLogger =
+            internalHeftSession.loggingManager.requestScopedLogger(`lifecycle:clean`);
+          cleanLogger.terminal.writeVerboseLine('Starting clean');
+
+          // Grab the additional clean operations from the phase
+          const deleteOperations: IDeleteOperation[] = [];
+
+          // Delete all temp folders for tasks by default
+          for (const pluginDefinition of lifecycle.pluginDefinitions) {
+            const { lifecycleSession } = await lifecycle.getContextForPluginDefinitionAsync(pluginDefinition);
+            deleteOperations.push({ sourcePath: lifecycleSession.tempFolder });
+
+            // Also delete the cache folder if requested
+            if (cleanCache) {
+              deleteOperations.push({ sourcePath: lifecycleSession.cacheFolder });
+            }
+          }
+
+          // Create the options and provide a utility method to obtain paths to delete
+          const cleanHookOptions: IHeftLifecycleCleanHookOptions = {
+            addDeleteOperations: (...deleteOperationsToAdd: IDeleteOperation[]) =>
+              deleteOperations.push(...deleteOperationsToAdd)
+          };
+
+          // Run the plugin clean hook
+          if (lifecycle.hooks.clean.isUsed()) {
+            try {
+              await lifecycle.hooks.clean.promise(cleanHookOptions);
+            } catch (e: unknown) {
+              // Log out using the clean logger, and return an error status
+              if (!(e instanceof AlreadyReportedError)) {
+                cleanLogger.emitError(e as Error);
+              }
+              return OperationStatus.Failure;
+            }
+          }
+
+          // Delete the files if any were specified
+          if (deleteOperations.length) {
+            await deleteFilesAsync(deleteOperations, cleanLogger);
+          }
+
+          cleanLogger.terminal.writeVerboseLine(`Finished clean (${performance.now() - startTime}ms)`);
+        }
+
+        // Run the start hook
+        if (lifecycle.hooks.toolStart.isUsed()) {
+          const lifecycleToolStartHookOptions: IHeftLifecycleToolStartHookOptions = {};
+          await lifecycle.hooks.toolStart.promise(lifecycleToolStartHookOptions);
+        }
+        break;
+      }
+      case 'stop': {
```

Comment:
```suggestion
      case 'end': {
```

Maybe "beginning" and "end"?

Reply:
Spoke while reviewing, changing to `start` and `finish`